File: src/database/models/Category.js
/**
 * Category Data Model
 */
export class Category {
  constructor(data = {}) {
    this.categoryId = data.categoryId || '';
    this.name = data.name || '';
    this.icon = data.icon || '';
    this.imageUrl = data.imageUrl || '';
    this.type = data.type || '';
    this.subCategories = data.subCategories || [];
    this.displayOrder = data.displayOrder || 0;
    this.isActive = data.isActive !== undefined ? data.isActive : true;
    this.createdAt = data.createdAt || null;
  }

  toJSON() {
    return {
      name: this.name,
      icon: this.icon,
      imageUrl: this.imageUrl,
      type: this.type,
      subCategories: this.subCategories,
      displayOrder: this.displayOrder,
      isActive: this.isActive,
      createdAt: this.createdAt
    };
  }
}

File: app/src/main/java/com/example/bunnix/database/models/Category.kt
package com.example.bunnix.database.models

import com.google.firebase.Timestamp
import com.google.firebase.firestore.DocumentId

data class Category(
    @DocumentId
    val categoryId: String = "",
    val name: String = "",
    val icon: String = "",
    val imageUrl: String = "",
    val type: String = "",
    val subCategories: List<String> = emptyList(),
    val displayOrder: Int = 0,
    val isActive: Boolean = true,
    val createdAt: Timestamp? = null
)